web/src/auth/LoginPage.js

```diff
@@ -133,6 +125,19 @@ class LoginPage extends React.Component {
     }
   }
 
+  updateCaptchaRule(application) {
+    const captchaProviderItems = this.getCaptchaProviderItems(application);
+    if (captchaProviderItems) {
+      if (captchaProviderItems.some(providerItem => providerItem.rule === "Always")) {
+        this.setState({enableCaptchaModal: CaptchaRule.Always});
```

Comment:
Why is `enableCaptchaModal` a React state from the beginning? Shouldn't it be a direct assignment like `this.getCaptchaRule(this.props.application);`?